tests/drivers/gpio/gpio_basic_api/testcase.yaml

```diff
@@ -30,3 +30,18 @@ tests:
     extra_args: "DTC_OVERLAY_FILE=boards/frdm_ke17z_fgpio.overlay"
     harness_config:
       fixture: fgpio_loopback
+
+  drivers.gpio.2pin_arduino:
+    min_flash: 34
+    depends_on:
+      - gpio
+      - arduino_gpio
+    filter: dt_compat_enabled("test-gpio-basic-api") and dt_compat_enabled("arduino-header-r3")
+    extra_args: "DTC_OVERLAY_FILE=arduino.overlay"
+    integration_platforms:
+      - frdm_k64f
+    platform_exclude:
+      # below boards are mikro bus not arduino
+      - da14695_dk_usb
+      - stm32h735g_disco
+      - stm32h750b_dk
```

Comment:
These boards actually feature a arduino connector, but its description is not provided.
But this is not important. My question is why would you need to put them in `platform_exclude` if filter `dt_compat_enabled("arduino-header-r3")` is effective ?

Reply:
the filter is for runtime, but as those boards define the `arduino_gpio`, in the board supported list, so it will build, and as there is not arduino dts defines, the build will fail.
```
supported:
  - arduino_gpio
```

Reply:
Ok, this should be cleaned up then. I'll take care of this.

Reply:
@erwango can you give your approval for this. Thanks again

Reply:
I propose to clean up these lines and rebase on top of https://github.com/zephyrproject-rtos/zephyr/pull/77729